test-harness/src/test/groovy/com/netflix/conductor/test/integration/SubWorkflowSpec.groovy

```diff
@@ -474,4 +488,1254 @@ class SubWorkflowSpec extends AbstractSpecification {
+
+        then: "verify that the 'sub_workflow_task' is in a IN_PROGRESS state"
+        def rootWorkflowInstance = workflowExecutionService.getExecutionStatus(rootWorkflowId, true)
+        with(rootWorkflowInstance) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 2
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.COMPLETED
+            tasks[1].taskType == TASK_TYPE_SUB_WORKFLOW
+            tasks[1].status == Task.Status.IN_PROGRESS
+        }
+
+        and: "verify that the mid-level workflow is RUNNING, and first task is in SCHEDULED state"
+        def midLevelWorkflowId = rootWorkflowInstance.tasks[1].subWorkflowId
+        with(workflowExecutionService.getExecutionStatus(midLevelWorkflowId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        and: "poll and complete the integration_task_1 task in the mid-level workflow"
+        workflowTestUtil.pollAndCompleteTask('integration_task_1', 'task1.integration.worker', ['op': 'task1.done'])
+
+        when: "the subworkflow task should be in SCHEDULED state and is started by issuing a system task call"
+        polledTaskIds = queueDAO.pop(TASK_TYPE_SUB_WORKFLOW, 1, 200)
+        workflowExecutor.executeSystemTask(WorkflowSystemTask.get(SubWorkflow.NAME), polledTaskIds.get(0), 30)
+        def midLevelWorkflowInstance = workflowExecutionService.getExecutionStatus(midLevelWorkflowId, true)
+
+        then: "verify that the mid-level workflow is RUNNING, and first task is in SCHEDULED state"
+        def leafWorkflowId = midLevelWorkflowInstance.tasks[1].subWorkflowId
+        def leafWorkflowInstance = workflowExecutionService.getExecutionStatus(leafWorkflowId, true)
+        with(leafWorkflowInstance) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        when: "poll and fail the integration_task_2 task"
+        workflowTestUtil.pollAndCompleteTask('integration_task_1', 'task1.integration.worker', ['op': 'task1.done'])
+        workflowTestUtil.pollAndFailTask('integration_task_2', 'task2.integration.worker', 'failed')
+
+        then: "the leaf workflow ends up in a FAILED state"
+        with(workflowExecutionService.getExecutionStatus(leafWorkflowId, true)) {
+            status == Workflow.WorkflowStatus.FAILED
+            tasks.size() == 2
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.COMPLETED
+            tasks[1].taskType == 'integration_task_2'
+            tasks[1].status == Task.Status.FAILED
+        }
+
+        when: "the mid level workflow is 'decided'"
```

Comment:
As noted earlier, this seems like a side-effect, since given current behavior, parent workflows are updated synchronously from the subworkflow. Sweep should not be needed.

Reply:
pls ignore